pypesto/visualize/sampling.py

```diff
@@ -91,7 +117,714 @@ def sampling_fval_trace(
+    )
+    plt.text(
+        xmin - artist_padding,
+        0.5,
+        'Simulated values',
+        ha='center',
+        va='center',
+        transform=fig.transFigure,
+        rotation='vertical'
+    )
+
+    # plt.tight_layout()  # Ruins layout for `groupby == OUTPUT`.
+    return axes
+
+
+def sampling_parameter_cis(
+        result: Result,
+        alpha: Sequence[int] = None,
+        step: float = 0.05,
+        show_median: bool = True,
+        title: str = None,
+        size: Tuple[float, float] = None,
+        ax: matplotlib.axes.Axes = None
+) -> matplotlib.axes.Axes:
+    """Plot MCMC-based parameter credibility intervals for one or more
+    credibility levels.
+
+    Parameters
+    ----------
+    result:
+        The pyPESTO result object with filled sample result.
+    alpha:
+        List of lower tail probabilities, defaults to 95% interval.
+    step:
+        Height of boxes for projectile plot, defaults to 0.05.
+    show_median:
+        Plot the median of the MCMC chain. Default: True.
+    title:
+        Axes title.
+    size: ndarray
+        Figure size in inches.
+    ax:
+        Axes object to use.
+
+    Returns
+    -------
+    ax:
+        The plot axes.
+    """
+    if alpha is None:
+        alpha = [95]
+
+    # automatically sort values in decreasing order
+    alpha_sorted = sorted(alpha, reverse=True)
+    # define colormap
+    evenly_spaced_interval = np.linspace(0, 1, len(alpha_sorted))
+    colors = [plt.cm.tab20c_r(x) for x in evenly_spaced_interval]
+    # number of sampled parameters
+    n_pars = result.sample_result.trace_x.shape[-1]
+
+    # set axes and figure
+    if ax is None:
+        _, ax = plt.subplots(figsize=size)
+
+    # loop over parameters
+    for npar in range(n_pars):
+        # initialize height of boxes
+        _step = step
+        # loop over confidence levels
+        for n, level in enumerate(alpha_sorted):
+            # extract percentile-based confidence intervals
+            lb, ub = calculate_ci(result=result, alpha=level / 100)
+
+            # assemble boxes for projectile plot
+            x1 = [lb[npar], ub[npar]]
+            y1 = [npar + _step, npar + _step]
+            y2 = [npar - _step, npar - _step]
+            # Plot boxes
+            ax.fill(np.append(x1, x1[::-1]),
+                    np.append(y1, y2[::-1]), color=colors[n],
+                    label=str(level) + '% CI')
+
+            if show_median:
+                if n == len(alpha_sorted) - 1:
+                    burn_in = result.sample_result.burn_in
+                    converged = result.sample_result.trace_x[0, burn_in:, npar]
+                    _median = np.median(converged)
+                    ax.plot([_median, _median], [npar - _step, npar + _step],
+                            'k-', label='MCMC median')
+
+            # increment height of boxes
+            _step += step
+
+    ax.set_yticks(range(n_pars))
+    ax.set_yticklabels(result.problem.x_names)
```

Comment:
This sets the first `n_par` parameters as yticklabels. In case you did not estimate all of them it can differ. Matplotlib >=3.3.0 even throws a ValueError (https://matplotlib.org/stable/api/api_changes.html#axis-set-ticklabels-must-match-fixedlocator-locs).
Instead `result.problem.get_reduced_vector(result.problem.x_names)` should be used as yticklabels.

Reply:
good point, thanks!